docs/hello_nextflow/03_hello_plumbing.md

```diff
@@ -0,0 +1,755 @@
+However, it is actually a symbolic link pointing to the the original file in the subdirectory of the first process call.
+By default, Nextflow uses symbolic links rather than copies to stage input and intermediate files.
+
+You'll also find the final outputs in the `results` directory since we used the `publishDir` directive in the second process too.
+
+!!! note
+
+    All we did was connect the output of `sayHello` to the input of `convertToUpper` and the two processes could be run serially.
+    Nextflow did the hard work of handling individual input and output files and passing them between the two commands for us.
+    This is the power of channels in Nextflow, doing the busywork of connecting our pipeline steps together.
+
+### Takeaway
+
+You know how to add a second step that takes the output of the first step as input.
+
+### What's next?
+
+Learn how to collect outputs from batched process calls and feed them into a single process.
+
+---
+
+## 2. Add a third step to collect all the greetings
+
+When we apply a transformation to a batch of inputs, like we're doing here to the multiple greetings, we'll often want to collect the transformed outputs and feed them into a single step that performs some kind of analysis or summation.
+
+Here we're simply going to write them all out to a single file, using the UNIX `cat` command.
+
+### 2.0. Run the collection command in the terminal
+
+The collection step we want to add to our workflow will use the `cat` command to concatenate multiple uppercased greetings into a single file.
+
+Let's run the command by itself in the terminal to verify that it works as expected, just like we've done previously.
+
+Run the following in your terminal:
+
+```bash
+echo 'Hello' | tr '[a-z]' '[A-Z]' > UPPER-Hello-output.txt
+echo 'Bonjour' | tr '[a-z]' '[A-Z]' > UPPER-Bonjour-output.txt
+echo 'Holà' | tr '[a-z]' '[A-Z]' > UPPER-Holà-output.txt
+cat UPPER-Hello-output.txt UPPER-Bonjour-output.txt UPPER-Holà-output.txt > COLLECTED-output.txt
+```
+
+The output is a text file called `COLLECTED-output.txt` that contains the uppercase versions of the original greetings.
+
+```console title="COLLECTED-output.txt"
+HELLO
+BONJOUR
+HOLà
+```
+
+That is the result we want to achieve with our workflow.
+
+### 2.1. Outline the collection step as a Nextflow process
+
+We can write an outline for our new process based on the previous one, leaving out a few pieces that require extra work.
+
+Add the following process definition to the workflow script:
+
+```groovy title="hello-plumbing.nf" linenums="41"
+/*
+ * Collect uppercase greetings into a single output file
+ */
+process collectGreetings {
+
+    publishDir 'results', mode: 'copy'
+
+    input:
+        ???
+
+    output:
+        path "COLLECTED-output.txt"
+
+    script:
+    """
+    ??? > 'COLLECTED-output.txt'
+    """
+}
+```
+
+This is what we can write based on what you've learned so far.
```

Comment:
Is this really true? We already know about variable path inputs, and the syntax isn't different for multiple files.

Reply:
Well we don't know yet that the syntax isn't different. As a naive user, I'm like ok, I'm confident in all of this, but for this next bit I need to check if the syntax is the same or if i need to do something special.